File: backend/employee/api/src/main/java/br/com/systec/taskflow/employee/api/service/EmployeeService.java
package br.com.systec.taskflow.employee.api.service;

import br.com.systec.taskflow.employee.api.exceptions.EmployeeFindException;
import br.com.systec.taskflow.employee.api.exceptions.EmployeeNotFoundException;
import br.com.systec.taskflow.employee.api.exceptions.EmployeeSavedException;
import br.com.systec.taskflow.employee.api.filter.EmployeeFilter;
import br.com.systec.taskflow.employee.api.vo.EmployeeVO;
import org.springframework.data.domain.Page;

public interface EmployeeService {

    EmployeeVO save(EmployeeVO employeeVO) throws EmployeeSavedException;

    EmployeeVO update(EmployeeVO employeeVO) throws EmployeeSavedException;

    EmployeeVO findById(Long id) throws EmployeeNotFoundException;

    Page<EmployeeVO> findByFilter(EmployeeFilter filter) throws EmployeeFindException;

    EmployeeVO findByUserId(Long userId) throws EmployeeFindException;
}

File: frontend/src/administrator/employee/service/employeeService.ts
import Page from "@/@core/utils/page";
import Employee from "@/administrator/employee/model/employee";
import { http } from "@/plugins/axios/axios";

export default class EmployeeService {
    private _endpoint = "/api/v1/employees";
    constructor() {}

    public async save(employee: Employee): Promise<Employee> {
        console.log(employee);
        if (employee.id === null) {
            return await this.create(employee);
        }

        return await this.update(employee);
    }

    private async create(employee: Employee): Promise<Employee> {
        console.log(employee);
        const response = await http.post<Employee>(
            this._endpoint,
            employee.toJson(),
        );

        return response.data;
    }

    private async update(employee: Employee): Promise<Employee> {
        const response = await http.put<Employee>(
            `${this._endpoint}/${employee.id}`,
            employee,
        );

        return response.data;
    }

    public async findById(id: number): Promise<Employee> {
        const response = await http.get<Employee>(`${this._endpoint}/${id}`);

        return response.data;
    }

    public async findByFilter(): Promise<Page<Employee>> {
        const response = await http.get<Page<Employee>>(
            `${this._endpoint}/filter`,
        );

        return response.data;
    }
}
